OpenCV build break for WinRT applications

##### System information (version)
- OpenCV => 3.2
- Operating System / Platform => Windows UWP/WinRT 64 Bit
- Compiler => Visual Studio 2015
- Windows 10 SDK 10.0.14393

##### Detailed description

OpenCV builds fail for Windows UWP/WinRT 64 bit.

##### Steps to reproduce
Build Visual Studio project and solution using following command line from builds\x64 directory:
"c:\Program Files\CMake\bin\cmake.exe" -G "Visual Studio 14 2015 Win64" -DCMAKE_SYSTEM_NAME=WindowsStore -DCMAKE_SYSTEM_VERSION=10.0 ..\..

Build resulting solution OpenCV.sln - ALL_BUILD in visual studio.

```
20>------ Build started: Project: opencv_highgui, Configuration: Debug x64 ------
20>  Building Custom Rule D:/GitHub/neilsh-msft/opencv/modules/highgui/CMakeLists.txt
20>  CMake does not need to re-run because D:/GitHub/neilsh-msft/opencv/Builds/x64/modules/highgui/CMakeFiles/generate.stamp is up-to-date.
20>  window.cpp
20>  roiSelector.cpp
20>  window_winrt.cpp
20>  window_winrt_bridge.cpp
20>D:\GitHub\neilsh-msft\opencv\modules\highgui\include\opencv2\highgui\highgui_winrt.hpp(44): error C2144: syntax error: 'void' should be preceded by ';' (compiling source file D:\GitHub\neilsh-msft\opencv\modules\highgui\src\window_winrt_bridge.cpp)
...
20>D:\GitHub\neilsh-msft\opencv\modules\highgui\src\window.cpp(225): error C3861: 'getenv': identifier not found

19>------ Build started: Project: opencv_videoio, Configuration: Debug x64 ------
19>  cap.cpp
19>  cap_images.cpp
19>  cap_mjpeg_encoder.cpp
19>  cap_mjpeg_decoder.cpp
19>  cap_winrt_capture.cpp
19>  cap_winrt_bridge.cpp
19>  cap_winrt_video.cpp
19>  CaptureFrameGrabber.cpp
19>D:\GitHub\neilsh-msft\opencv\modules\videoio\src\cap_winrt_bridge.cpp(84): warning C4458: declaration of 'width' hides class member
...
19>D:\GitHub\neilsh-msft\opencv\modules\videoio\src\cap_winrt\CaptureFrameGrabber.cpp(100): error C2653: 'CameraOptionsUI': is not a class or namespace name
19>D:\GitHub\neilsh-msft\opencv\modules\videoio\src\cap_winrt\CaptureFrameGrabber.cpp(100): error C3861: 'Show': identifier not found
```
The first break is because window_winrt_bridge first includes highgui_winrt.hpp without including any of the opencv common definitions.  The second break in window.cpp is because window.cpp is using undeclared HAVE_WINRT to ifndef out the getenv() operation instead of the proper WINRT macro.  The final break is because videoio does not reference the WinRT desktop extensions
Fixes: 
```
--- a/modules/highgui/src/window_winrt_bridge.cpp
+++ b/modules/highgui/src/window_winrt_bridge.cpp
@@ -24,6 +24,8 @@
 // NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
 // POSSIBILITY OF SUCH DAMAGE.

+#include "precomp.hpp"
+
 #include "opencv2\highgui\highgui_winrt.hpp"
 #include "window_winrt_bridge.hpp"

--- a/modules/highgui/src/window.cpp
+++ b/modules/highgui/src/window.cpp
@@ -218,7 +218,7 @@ int cv::waitKey(int delay)
 {
     CV_TRACE_FUNCTION();
     int code = waitKeyEx(delay);
-#ifndef HAVE_WINRT
+#ifndef WINRT
     static int use_legacy = -1;
     if (use_legacy < 0)
     {

--- a/modules/videoio/CMakeLists.txt
+++ b/modules/videoio/CMakeLists.txt
@@ -256,6 +256,9 @@ endif()

 if(MSVC)
   set_target_properties(${the_module} PROPERTIES LINK_FLAGS "/NODEFAULTLIB:atlthunk.lib /NODEFAULTLIB:atlsd.lib /NODEFAULTLIB:libcmt.lib /DEBUG")
+  if(DEFINED WINRT AND NOT DEFINED ENABLE_WINRT_MODE_NATIVE)
+    set_target_properties(${the_module} PROPERTIES VS_DESKTOP_EXTENSIONS_VERSION "${CMAKE_VS_WINDOWS_TARGET_PLATFORM_VERSION}")
+  endif()
 endif()

```

--- a/modules/highgui/src/window.cpp
+++ b/modules/highgui/src/window.cpp
@@ -218,7 +218,7 @@ int cv::waitKey(int delay)
 {
     CV_TRACE_FUNCTION();
     int code = waitKeyEx(delay);
-#ifndef HAVE_WINRT
+#ifndef WINRT
     static int use_legacy = -1;
     if (use_legacy < 0)
     {

--- a/modules/highgui/src/window_winrt_bridge.cpp
+++ b/modules/highgui/src/window_winrt_bridge.cpp
@@ -24,6 +24,8 @@
 // NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
 // POSSIBILITY OF SUCH DAMAGE.
 
+#include "precomp.hpp"
+
 #include "opencv2\highgui\highgui_winrt.hpp"
 #include "window_winrt_bridge.hpp"
 

--- a/modules/videoio/CMakeLists.txt
+++ b/modules/videoio/CMakeLists.txt
@@ -256,6 +256,9 @@ endif()
 
 if(MSVC)
   set_target_properties(${the_module} PROPERTIES LINK_FLAGS "/NODEFAULTLIB:atlthunk.lib /NODEFAULTLIB:atlsd.lib /NODEFAULTLIB:libcmt.lib /DEBUG")
+  if(DEFINED WINRT AND NOT DEFINED ENABLE_WINRT_MODE_NATIVE)
+    set_target_properties(${the_module} PROPERTIES VS_DESKTOP_EXTENSIONS_VERSION "${CMAKE_VS_WINDOWS_TARGET_PLATFORM_VERSION}")
+  endif()
 endif()
 
 #stop automatic dependencies propagation for this module

